scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -537,8 +537,14 @@ class FossId internal constructor(
         if (response.data?.status in SCAN_STATE_FOR_TRIGGER) {
             log.info { "Triggering scan as it has not yet been started." }
 
-            service.runScan(config.user, config.apiKey, scanCode, *runOptions)
-                .checkResponse("trigger scan", false)
+            val scanResult = service.runScan(config.user, config.apiKey, scanCode, *runOptions)
+
+            // Scans that were added to the queue are interpreted as an error by FossID before version 2021.1.
+            // For older versions, `waitScanComplete()` is able to deal with queued scans. Therefore, not checking the
+            // response of queued scans.
+            if (version >= "2021.1" || scanResult.error != "Scan was added to queue") {
```

Comment:
Does `>=` on a string really do what we want here? Does it work because it's CalVer? Just want to double-check 😀 

Anyway, this probably also need to say `"2021.2"` (see above).

Reply:
Yes it does what we want, because it's CalVer.
This is also how it is done in https://github.com/oss-review-toolkit/ort/blob/cf5efc563cc5e4ede8e4c79d351d8d1d00621776/clients/fossid-webapp/src/main/kotlin/FossIdServiceWithVersion.kt#L36

Thanks for the hint with the wrong version.